### ISSUE TITLE
hl.hadoop_stat() returning wrong type for modification_time using Google Cloud Storage filesystem

### ISSUE THREAD

**Issue by mattsolo1** [broadinstitute/gnomad-browser#915]
```
Welcome to
     __  __     <>__
    / /_/ /__  __/ /
   / __  / _ `/ / /
  /_/ /_/\_,_/_/_/   version 0.2.95-513139587f57
LOGGING: writing to /home/hail/hail-20220621-2016-0.2.95-513139587f57.log
2022-06-21 20:16:51,226 - gnomad_data_pipeline - INFO - Skipping download_gencode_v19_gtf
2022-06-21 20:16:51,254 - gnomad_data_pipeline - INFO - Skipping download_gencode_v35_gtf
2022-06-21 20:16:51,276 - gnomad_data_pipeline - INFO - Skipping download_hgnc
Traceback (most recent call last):
  File "/tmp/d6e9966b27524ef4acf8779e1439d7c1/genes.py", line 325, in <module>
    run_pipeline(pipeline)
  File "/tmp/d6e9966b27524ef4acf8779e1439d7c1/pyfiles_as0u9erm.zip/data_pipeline/pipeline.py", line 194, in run_pipeline
  File "/tmp/d6e9966b27524ef4acf8779e1439d7c1/pyfiles_as0u9erm.zip/data_pipeline/pipeline.py", line 161, in run
  File "/tmp/d6e9966b27524ef4acf8779e1439d7c1/pyfiles_as0u9erm.zip/data_pipeline/pipeline.py", line 122, in run
  File "/tmp/d6e9966b27524ef4acf8779e1439d7c1/pyfiles_as0u9erm.zip/data_pipeline/pipeline.py", line 112, in should_run
  File "/tmp/d6e9966b27524ef4acf8779e1439d7c1/pyfiles_as0u9erm.zip/data_pipeline/pipeline.py", line 48, in modified_time
  File "/tmp/d6e9966b27524ef4acf8779e1439d7c1/pyfiles_as0u9erm.zip/data_pipeline/pipeline.py", line 27, in modified_time
TypeError: strptime() argument 1 must be str, not float
```

The Hail docs say that stat["modification_time"] should be a string, but in the case of Google Cloud Storage, it returns an epoch timestamp as an int.

See https://github.com/broadinstitute/gnomad-browser/pull/910

**Comment by mattsolo1** [comment]
I understand now that this issue only arises when re-running pipelines.

**Comment by phildarnowsky-broad** [comment]
@mattsolo1 correct, the pipeline as it stands only calls the problematic method if there are intermediate results left behind by a previous run of the pipeline.

**Comment by mattsolo1** [comment]
@phildarnowsky-broad is this now fixed by #910?
